test/Confluent.Kafka.IntegrationTests/Tests/Builder_Handlers.cs

```diff
@@ -0,0 +1,235 @@
+            {
+                GroupId = Guid.NewGuid().ToString(),
+                BootstrapServers = bootstrapServers,
+                SessionTimeoutMs = 6000,
+                EnablePartitionEof = true,
+                Debug = "all"
+            };
+
+            ManualResetEventSlim mres1 = new(), mres2 = new();
+
+            using var consumer = new ConsumerBuilder<byte[], byte[]>(consumerConfig)
+                .SetPartitionsAssignedHandler((c, partitions) =>
+                {
+                    Assert.Single(partitions);
+                    Assert.Equal(firstProduced.TopicPartition, partitions[0]);
+                    return partitions.Select(p => new TopicPartitionOffset(p, firstProduced.Offset));
+                })
+                .SetLogHandler((_, _) => mres1.Set())
+                .SetLogHandler((_, _) => mres2.Set())
+                .Build();
+            consumer.Subscribe(singlePartitionTopic);
+
+            Assert.True(mres1.Wait(TimeSpan.FromSeconds(5)));
+            Assert.True(mres2.Wait(TimeSpan.FromSeconds(5)));
+
+            LogToFile("end   ConsumerBuilder_SetLogHandler");
+        }
+        
+        [Theory, MemberData(nameof(KafkaParameters))]
+        public void ConsumerBuilder_SetStatisticsHandler(string bootstrapServers)
+        {
+            LogToFile("start ConsumerBuilder_SetStatisticsHandler");
+            
+            int N = 2;
+            var firstProduced = Util.ProduceNullStringMessages(bootstrapServers, singlePartitionTopic, 100, N);
+            
+            var consumerConfig = new ConsumerConfig
+            {
+                GroupId = Guid.NewGuid().ToString(),
+                BootstrapServers = bootstrapServers,
+                SessionTimeoutMs = 6000,
+                EnablePartitionEof = true,
+                StatisticsIntervalMs = 100
+            };
+
+            ManualResetEventSlim mres1 = new(), mres2 = new();
+
+            using (var consumer = new ConsumerBuilder<byte[], byte[]>(consumerConfig)
+                       .SetPartitionsAssignedHandler((c, partitions) =>
+                       {
+                           Assert.Single(partitions);
+                           Assert.Equal(firstProduced.TopicPartition, partitions[0]);
+                           return partitions.Select(p => new TopicPartitionOffset(p, firstProduced.Offset));
+                       })
+                       .SetStatisticsHandler((_, _) => mres1.Set())
+                       .SetStatisticsHandler((_, _) => mres2.Set())
+                       .Build())
+            {
+                consumer.Subscribe(singlePartitionTopic);
+                
+                int msgCnt = 0;
+                while (true)
+                {
+                    var record = consumer.Consume(TimeSpan.FromMilliseconds(100));
+                    if (record == null) { continue; }
+                    if (record.IsPartitionEOF) { break; }
+                    msgCnt += 1;
```

Comment:
Do you want to assert how many messages were counted?

Reply:
No that's not required here. good catch, I dropped the useless `msgCnt` variables.